### PR TITLE
(PCP-338) Handle Associate Session errors

### PR DESCRIPTION
**Pull request by parisiale**
Retrying to connect in case of errors during Session Association; that
is done after a random pause in the  [5 s, 10 s] interval.

Handling Session Association failures.

### CODE REVIEW

**Comment by parisiale** [comment]
Don't merge before https://github.com/puppetlabs/cpp-pcp-client/pull/144/files.

File: exe/main.cc [hunk 1]
```diff
@@ -66,9 +66,12 @@ int startAgent(std::vector<std::string> arguments) {
     try {
         Agent agent { Configuration::Instance().getAgentConfiguration() };
         agent.start();
+    } catch (const Agent::PCPConfigurationError& e) {
```
Comment (james-stocks) [inline exe/main.cc]: This new Error struct is used in the event that association fails - should it therefore be named PCPAssociationError or something similar? To a user or a support engineer; "PCP configuration" might suggest something in a config file. I think in this context "configuration" instead means the combination of PCP broker/client components?

Comment (parisiale) [inline exe/main.cc]: No.

**Comment by mruzicka** [comment]
LGTM, although I agree with @james-stocks, the exception naming could use some improvement.

**Pull request merged**